src/main/scala/org/broadinstitute/dsde/firecloud/utils/TSVFormatter.scala

```diff
@@ -153,11 +153,13 @@ object TSVFormatter {
     * @return Ordered list of rows, each row entry value ordered by its corresponding header position
     */
   def makeEntityRows(entityType: String, entities: Seq[Entity], headers: IndexedSeq[String]): IndexedSeq[IndexedSeq[String]] = {
-    val memberPlural = pluralizeMemberType(memberTypeFromEntityType(entityType))
     // if we have a set entity, we need to filter out the attribute array of the members so that we only
     // have top-level attributes to construct columns from.
-    val filteredEntities = if (isCollectionType(entityType)) {
-      filterAttributeFromEntities(entities, memberPlural)
+    val filteredEntities = if (FlexibleModelSchema.isCollectionType(entityType)) {
+      FlexibleModelSchema.memberTypeFromEntityType(entityType) match {
+        case Success(memberType) => filterAttributeFromEntities(entities, FlexibleModelSchema.pluralizeEntityType(memberType))
+        case Failure(regret) => ??? // should not happen?
```

Comment:
any suggestions on what to do here? throw exception?

Reply:
Maybe a 500 error, yeah

Reply:
It can happen if the collection type is not named with a `_set` suffix which, outside of the FC data model, is a user error. Therefore, I think just calling `.get` instead of matching and letting the exception be thrown would be fine.
```
    val filteredEntities = if (FlexibleModelSchema.isCollectionType(entityType)) {
      val memberPlural = FlexibleModelSchema.pluralizeEntityType(FlexibleModelSchema.memberTypeFromEntityType(entityType).get)
      filterAttributeFromEntities(entities, memberPlural)
    } else {
      entities
    }
```